### PR TITLE
Update to MV3: DeclarativeContent API 

### PR DESCRIPTION
**Pull request by AmySteam**
Changes proposed in this pull request:

- Convert references to pageaction into action (MV3)
- Add link to [Emulating pageActions with declarativeContent](https://developer.chrome.com/docs/extensions/reference/action/#emulating-pageactions-with-declarativecontent)
- Convert Numbered links to text links
- Clarify how to use this API.

[Staged link](https://deploy-preview-2261--developer-chrome-com.netlify.app/docs/extensions/reference/declarativeContent)

### CODE REVIEW

**Review by dotproto — CHANGES_REQUESTED** [review]
Looking good. I think after the requested changes land we're good to tag in Joe.

**Comment by AmySteam** [comment]
@jpmedley This is ready for an editorial review. Thanks!

**Comment by jpmedley** [comment]
@dotproto Are you done?

**Review by dotproto — APPROVED** [review]
LGTM

**Comment by dotproto** [comment]
@jpmedley, this is ready for you

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 1]
```diff
@@ -2,26 +2,31 @@
...
+{% endAside %}
+
 ## Usage
 
-The Declarative Content API allows you to show your extension's [page action][1] depending on the
-URL of a web page and the CSS selectors its content matches, without needing to take a [host
-permission][2] or inject a [content script][3]. Use the [activeTab][4] permission in order to be
-able to interact with a page after the user clicks on your page action.
+The Declarative Content API allows you to enable your extension's [action][api-action] depending on
+the URL of a web page and the CSS selectors its content matches, without needing to add [host
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:14]: This phrasing is a bit odd. Does it mean that a URL triggers an action?

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 2]
```diff
@@ -2,26 +2,31 @@
...
+permission][docs-host-perm] or inject a [content script][docs-content-scripts]. 
 
-If you need more precise control over when your page action appears or you need to change its
-appearance to match the current tab before the user clicks on it, you'll have to keep using the
-[pageAction][5] API.
+Use the [activeTab][docs-activetab] permission to interact with a page after the user clicks on the
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:17]:
```suggestion
Use the [activeTab][docs-activetab] permission to interact with a page after the user clicks the
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 3]
```diff
@@ -2,26 +2,31 @@
...
+extension's action (toolbar icon).
 
 ## Rules
 
-As a [declarative API][6], this API lets you register rules on the [`onPageChanged`][7] [event][8]
-object which take an action ([`ShowPageAction`][9] and [`SetIcon`][10]) when a set of conditions,
-represented as a [`PageStateMatcher`][11], are met.
+As a [declarative API][api-declarative], this API lets you register rules on the
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:22]:
```suggestion
The [declarative API][api-declarative] lets you register rules on the
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 4]
```diff
@@ -2,26 +2,31 @@
...
+[`onPageChanged`][event-onpagechanged] event object which take an action
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:23]: >lets you register rules on the on the `onPageChanged` event object

This phrase feels like its cramming too much in formation in too small a space and I can't disentangle it. I would expect something called `onPageChanged` to be defined on a target object, take a custom method (a callback, though it's not usually called that in JS) and for the method to get passed an event object. This description reads like it's conflated the target, the callback, and the event into one thing. It's confusing.

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: site/en/docs/extensions/reference/declarativeContent/index.md (hunk 4 above)
Comment (AmySteam) [reply site/en/docs/extensions/reference/declarativeContent/index.md]: Agreed! I restructured this section to detangle key concepts. Let me know what you think.

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 5]
```diff
@@ -2,26 +2,31 @@
...
+
 ## Usage
 
-The Declarative Content API allows you to show your extension's [page action][1] depending on the
-URL of a web page and the CSS selectors its content matches, without needing to take a [host
-permission][2] or inject a [content script][3]. Use the [activeTab][4] permission in order to be
-able to interact with a page after the user clicks on your page action.
+The Declarative Content API allows you to enable your extension's [action][api-action] depending on
+the URL of a web page and the CSS selectors its content matches, without needing to add [host
+permission][docs-host-perm] or inject a [content script][docs-content-scripts].
```
[the remark this answers is not in the document]
Comment (AmySteam) [reply site/en/docs/extensions/reference/declarativeContent/index.md]: Yes! Good catch :)

**Comment by AmySteam** [comment]
@jpmedley I made a few changes to explain this API better. Can you review it again? Thank you!

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 6]
```diff
@@ -2,64 +2,79 @@
...
+
 ## Usage
 
-The Declarative Content API allows you to show your extension's [page action][1] depending on the
-URL of a web page and the CSS selectors its content matches, without needing to take a [host
-permission][2] or inject a [content script][3]. Use the [activeTab][4] permission in order to be
-able to interact with a page after the user clicks on your page action.
+{% Aside 'key-term' %}
+
+An [_extension's action_][api-action] controls the extension's toolbar icon.
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:16]:
```suggestion
An [extension's _action_][api-action] controls the extension's toolbar icon.
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 7]
```diff
@@ -2,64 +2,79 @@
...
+
+{% endAside %}
+
+The Declarative Content API allows you to enable your extension's action depending on the URL of a
+web page and the CSS selectors, without needing to add [host permissions][docs-host-perm] or inject a
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:21]: >and the CSS selectors.

What is this telling me? I don't understand this part.

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 8]
```diff
@@ -2,64 +2,79 @@
...
+{% Aside 'success' %}
 
-**Note:** All conditions and actions are created via a constructor as shown in the example above.
+All conditions and actions are created via a constructor as shown in the example above.
 
 {% endAside %}
 
-In order to also show a page action for sites with a video, you can add a second condition, as each
-condition is sufficient to trigger all specified actions:
+In order to also enable the extension's action for Google sites with a video, you can add a second
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:56]:
```suggestion
To also enable the extension's action for Google sites with a video, you can add a second
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 9]
```diff
@@ -2,64 +2,79 @@
...
     })
   ],
-  actions: [ new chrome.declarativeContent.ShowPageAction() ]
+  actions: [ new chrome.declarativeContent.ShowAction() ]
 };
 ```
 
-[Added rules][27] are saved across browser restarts, so register them as follows:
+The [`onPageChanged`][event-onpagechanged] event tests whether any rule has at least one fulfilled
+condition and executes the actions. Rules are persistent across browsing sessions; therefore, during
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:75]:
```suggestion
condition and executes the actions. Rules persist across browsing sessions; therefore, during
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 10]
```diff
@@ -71,70 +86,91 @@ chrome.runtime.onInstalled.addListener(function(details) {
...
-include [combinators][36] like whitespace or "`>`" in your selectors. This helps Chrome match the
-selectors more efficiently.
-
-<table><tbody><tr><th>Compound Selectors (OK)</th><th>Complex Selectors (Not OK)</th></tr><tr><td><code>a</code></td><td><code>div p</code></td></tr><tr><td><code>iframe.special[src^='http']</code></td><td><code>p&gt;span.highlight</code></td></tr><tr><td><code>ns|*</code></td><td><code>p + ol</code></td></tr><tr><td><code>#abcd:checked</code></td><td><code>p::first-line</code></td></tr></tbody></table>
+With the [activeTab][docs-activetab] permission, your extension will not display any permission
+warnings and when the user clicks the extension action, it will only run on relevant pages.
+
+## Page URL Matching {: #page-url}
+
+The [`PageStateMatcher.pageurl`][type-matcher-url] matches if the URL criterias are fulfilled. The
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:100]:
```suggestion
The [`PageStateMatcher.pageurl`][type-matcher-url] matches when the URL criteria are fulfilled. The
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 11]
```diff
@@ -71,70 +86,91 @@ chrome.runtime.onInstalled.addListener(function(details) {
...
+most common criterias are a concatination of either host, path, or url, followed by Contains, Equals, Prefix, or
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:101]:
```suggestion
most common criteria are a concatenation of either host, path, or url, followed by Contains, Equals, Prefix, or
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 12]
```diff
@@ -71,70 +86,91 @@ chrome.runtime.onInstalled.addListener(function(details) {
...
+Suffix. The following table contains a few examples:
+
+| Criteria                                  | Matches                         |
+|-------------------------------------------|---------------------------------|
+| `{ hostSuffix: 'google.com' }`            | All Google URLs                 |
+| `{ pathPrefix: '/docs/extensions'` }      | Extension docs URLs             |
+| `{ urlContains: 'developer.chrome.com'` } | All chrome developers docs URLs |
+
+All criteria are case sensitive. For complete list of criterias, see [UrlFilter][type-urlfilter].
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:110]:
```suggestion
All criteria are case sensitive. For a complete list of criteria, see [UrlFilter][type-urlfilter].
```

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 13]
```diff
@@ -71,70 +86,91 @@ chrome.runtime.onInstalled.addListener(function(details) {
...
+
+## CSS Matching {: #css}
+
+[`PageStateMatcher.css`][type-matcher-css] conditions must be _[comp``ound selectors][w3-compound]_,
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/declarativeContent/index.md:114]:
```suggestion
[`PageStateMatcher.css`][type-matcher-css] conditions must be _[compound selectors][w3-compound]_,
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: site/en/docs/extensions/reference/declarativeContent/index.md [hunk 14]
```diff
@@ -2,64 +2,79 @@
...
-permission][2] or inject a [content script][3]. Use the [activeTab][4] permission in order to be
-able to interact with a page after the user clicks on your page action.
+{% Aside 'key-term' %}
+
+An [extension's _action_][api-action] controls the extension's toolbar icon.
+
+{% endAside %}
+
+The Declarative Content API allows you to enable your extension's action depending on the URL of a
+web page, or if a CSS selector matches an element on the page, without needing to
```
Comment (AmySteam) [inline site/en/docs/extensions/reference/declarativeContent/index.md:21]: @jpmedley How does this sound?

File: site/en/docs/extensions/reference/declarativeContent/index.md (hunk 7 above)
Comment (AmySteam) [reply site/en/docs/extensions/reference/declarativeContent/index.md]: I just made a change, let me know what you think

File: site/en/docs/extensions/reference/declarativeContent/index.md (hunk 14 above)
Comment (jpmedley) [reply site/en/docs/extensions/reference/declarativeContent/index.md:21]: Sounds good.

**Pull request merged**